x-pack/plugins/security/server/config.test.ts

```diff
@@ -0,0 +1,432 @@
+    const contextMock = coreMock.createPluginInitializerContext({
+      encryptionKey: 'a'.repeat(32),
+      secureCookies: false,
+      authc: { providers: ['basic'] },
+    });
+
+    const config = await createConfig$(contextMock, false)
+      .pipe(first())
+      .toPromise();
+    expect(config).toMatchInlineSnapshot(`
+            Object {
+              "authc": Object {
+                "providers": Array [
+                  "basic",
+                ],
+              },
+              "encryptionKey": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+              "secureCookies": false,
+            }
+        `);
+
+    expect(loggingServiceMock.collect(contextMock.logger).warn).toMatchInlineSnapshot(`
+                              Array [
+                                Array [
+                                  "Session cookies will be transmitted over insecure connections. This is not recommended.",
+                                ],
+                              ]
+                    `);
+  });
+
+  it('should log a warning if SSL is not configured yet secure cookies are being used', async () => {
+    const contextMock = coreMock.createPluginInitializerContext({
+      encryptionKey: 'a'.repeat(32),
+      secureCookies: true,
+      authc: { providers: ['basic'] },
+    });
+
+    const config = await createConfig$(contextMock, false)
+      .pipe(first())
+      .toPromise();
+    expect(config).toMatchInlineSnapshot(`
+            Object {
+              "authc": Object {
+                "providers": Array [
+                  "basic",
+                ],
+              },
+              "encryptionKey": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+              "secureCookies": true,
+            }
+        `);
+
+    expect(loggingServiceMock.collect(contextMock.logger).warn).toMatchInlineSnapshot(`
+                              Array [
+                                Array [
+                                  "Using secure cookies, but SSL is not enabled inside Kibana. SSL must be configured outside of Kibana to function properly.",
+                                ],
+                              ]
+                    `);
+  });
+
+  it('should set xpack.security.secureCookies if SSL is configured', async () => {
+    const contextMock = coreMock.createPluginInitializerContext({
+      encryptionKey: 'a'.repeat(32),
+      secureCookies: false,
+      authc: { providers: ['basic'] },
+    });
+
+    const config = await createConfig$(contextMock, true)
+      .pipe(first())
+      .toPromise();
+    expect(config).toMatchInlineSnapshot(`
+      Object {
+        "authc": Object {
+          "providers": Array [
+            "basic",
+          ],
+        },
+        "encryptionKey": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        "secureCookies": true,
+      }
+    `);
+
+    expect(loggingServiceMock.collect(contextMock.logger).warn).toEqual([]);
+  });
+
+  it('should set authProviders to authc.providers', async () => {
```

Comment:
__note:__ diff#11: added test for the "rename deprecation transformation" ugly hacky workaround I had to add in this backport (see below).